pyop2/configuration.py

```diff
@@ -40,6 +40,22 @@
 from pyop2.exceptions import ConfigurationError
 
 
+def default_simd_width():
+    from cpuinfo import get_cpu_info
+    avx_to_width = {'avx': 2, 'avx1': 2, 'avx128': 2, 'avx2': 4,
+                    'avx256': 4, 'avx3': 8, 'avx512': 8}
+    longest_ext = [t for t in get_cpu_info()["flags"] if t.startswith('avx')][-1]
+    if longest_ext not in avx_to_width.keys():
+        if longest_ext[:6] not in avx_to_width.keys():
+            assert longest_ext[:4] in avx_to_width.keys(), \
```

Comment:
Users should never see assertion errors. Can you replace with [`ConfigurationError`](https://github.com/OP2/PyOP2/blob/master/pyop2/exceptions.py#L136) or, probably better, just fall back to a non-vectorised solution.